client/internal/dns/upstream.go

```diff
@@ -172,6 +174,11 @@ func (u *upstreamResolverBase) probeAvailability() {
 	default:
 	}
 
+	// avoid probe if upstreams could resolve at least one query and fails count is less than failsTillDeact
+	if u.successCount.Load() > 0 && u.failsCount.Load() > u.failsTillDeact {
```

Comment:
shouldn't this be
```suggestion
	if u.successCount.Load() > 0 && u.failsCount.Load() < u.failsTillDeact {
```
?

Besides, that would prevent us from failing that upstream dns server if it became unavaible later, correct?

Reply:
No, as the counters are updated according to their state